### PR TITLE
build: drop CentOS 7 builds

### PR DESCRIPTION
**Pull request by howardjohn**
This support was never fully launched, but causes a large supportabiltiy burden on the proxy. Removing this.

Related: https://github.com/istio/tools/pull/2549 https://github.com/istio/proxy/pull/4758 https://github.com/istio/test-infra/pull/4860 https://github.com/istio/istio/pull/45735

### CODE REVIEW

File: pkg/test/framework/components/echo/config.go [hunk 1]
```diff
@@ -73,7 +73,6 @@ const (
 	UbuntuXenial VMDistro = "UbuntuXenial"
 	UbuntuJammy  VMDistro = "UbuntuJammy"
 	Debian11     VMDistro = "Debian9"
```
Comment (kyessenov) [inline pkg/test/framework/components/echo/config.go:75]: Not related but this line has a typo Debian11 = Debian9

Comment (howardjohn) [reply pkg/test/framework/components/echo/config.go:75]: yikes... thanks. Ill send another PR

**Review by kyessenov — APPROVED** [review]

File: pkg/test/framework/components/echo/config.go (hunk 1 above)
Comment (howardjohn) [reply pkg/test/framework/components/echo/config.go:75]: Actually not as bad as I thought, I think its cosmetic. But still good to fix

**Review by ericvn — APPROVED** [review]

**Review by jacob-delgado — APPROVED** [review]

**Pull request merged**